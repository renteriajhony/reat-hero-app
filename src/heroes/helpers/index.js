// export * from './';
export * from './getHeroesById';
export * from './getHeroesByPublisher';
export * from './getHeroesByName';
