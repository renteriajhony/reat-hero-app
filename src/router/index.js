export * from './AppRouter';
export * from './AppRouteName';
export * from './PrivateRoute';
export * from './PublicRoute';
