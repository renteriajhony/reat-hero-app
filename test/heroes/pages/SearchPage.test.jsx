import { fireEvent, render, screen } from '@testing-library/react';
import { SearchPage } from '../../../src/heroes';
import { MemoryRouter } from 'react-router-dom';


const mockUseNavegate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavegate,
}));
describe('Test <SearchPage/>', () => {
  beforeEach(() => jest.clearAllMocks());
  beforeAll(() => {});
  afterAll(() => {});

  test('should render with values for defaulth', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    // screen.debug();
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Searching'));
  });
  test('Should show batman and the input with the parameter value', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=bat']}>
        <SearchPage />
      </MemoryRouter>
    );
    // screen.debug();
    const input = screen.getByRole('textbox');
    expect(input.value).toBe('bat');
    const img = screen.getByRole('img');
    expect(img.src).toContain('assets/heroes/dc-batman.jpg');

    const divAlertSearch = screen.getByLabelText('divAlertSearch');
    expect(divAlertSearch.style.display).toBe('none');
  });

  test('should show error when not finding the hero', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batara']}>
        <SearchPage />
      </MemoryRouter>
    );
    // screen.debug();

    const divAlertDanger = screen.getByLabelText('divAlertDanger');
    expect(divAlertDanger.style.display).toBe('block');
  });
  test('should call the navigator to the new screen', () => {
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );
    // screen.debug();

    const input = screen.getByRole('textbox');
    fireEvent.change(input, {
      target: { name: 'searchText', value: 'batman' },
    });
    const form = screen.getByLabelText('formSearch');
    fireEvent.submit(form);

    expect(mockUseNavegate).toHaveBeenCalledWith('?q=batman');
  });
});
