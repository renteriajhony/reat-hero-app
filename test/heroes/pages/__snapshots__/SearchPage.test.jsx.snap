// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SearchPage/> should render with values for defaulth 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="formSearch"
      >
        <input
          autocomplete="off"
          class="form-control"
          id="searchText"
          name="searchText"
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="divAlertSearch"
        class="alert alert-primary animate__animated animate__fadeIn"
        style="display: block;"
      >
        Search hero
      </div>
      <div
        aria-label="divAlertDanger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        Thereâ€™s no result for: 
        <b />
      </div>
    </div>
  </div>
</div>
`;
